### PR TITLE
RavenDB-13738 Have a better error why 4.2 can't talk with 4.1.4

### PR DESCRIPTION
**Pull request by karmeli87**

### CODE REVIEW

File: src/Raven.Server/Rachis/RachisConsensus.cs [hunk 1]
```diff
@@ -1112,6 +1116,15 @@ public void AcceptNewConnection(Stream stream, Action disconnect, EndPoint remot
             }
         }
 
+        private static void ValidateCompatibility(RachisHello initialMessage)
+        {
+            if (ServerVersion.IsNightlyOrDev(initialMessage.ServerBuildVersion) == false &&
+                initialMessage.ServerBuildVersion <= 41010)
```
Comment (ppekrol) [inline src/Raven.Server/Rachis/RachisConsensus.cs]: Old versions will not send anything so you will get 0. What will happen then?
Also 4.2.0 will not send anything. Actually nodes <= 41010 are not sending anything so I'm not sure if there is a point of having this at the moment? Can be used in future, but not know?

Comment (karmeli87) [reply src/Raven.Server/Rachis/RachisConsensus.cs]: you are correct, at all points :)

File: src/Raven.Server/Rachis/RachisConsensus.cs [hunk 2]
```diff
@@ -1112,6 +1116,13 @@ public void AcceptNewConnection(Stream stream, Action disconnect, EndPoint remot
             }
         }
 
+        private static void ValidateCompatibility(RachisHello initialMessage)
+        {
+            var version = initialMessage.ServerBuildVersion;
+            if (version == 0 || ServerVersion.IsNightlyOrDev(version))  
+                return;
+        }
```
Comment (ayende) [inline src/Raven.Server/Rachis/RachisConsensus.cs:1124]: What is the point of this? Preparing for the future?

Comment (ppekrol) [reply src/Raven.Server/Rachis/RachisConsensus.cs:1124]: Yes, correct. In theory we can remove this, but maybe it is better to have this extension point prepared.

**Review by ayende — APPROVED** [review]

**Pull request merged**